superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.h

```diff
@@ -0,0 +1,143 @@
+// it has no understanding of the lifetime of resources on the GPU. Apps must account
+// for the GPU lifetime of resources to avoid destroying objects that may still be
+// referenced by the GPU.
+// An example of this can be found in the class method: OnDestroy().
+using Microsoft::WRL::ComPtr;
+
+#define SAFE_RELEASE(p)                                                                                                \
+    if (p)                                                                                                             \
+    (p)->Release()
+template <typename T> T *get_rvalue_ptr(T &&v) { return &v; }
+
+class GPUCore {
+  public:
+    GPUCore(Options *opts) : opts(opts) {}
+    ~GPUCore() { FlushCommandQueue(); }
+
+    /**
+     * @brief Setup GPU and start benchmark.
+     */
+    void Run();
+
+    /**
+     * @brief Create pipeline including
+     *		  create device object, command list, command queue
+     *		  and synchronization objects.
+     */
+    void LoadPipeline();
+
+    /**
+     * @brief Prepare input and output data and buffers of the tensor elements..
+     */
+    template <typename T> void PrepareData(const int m, const int n, const int k);
+
+    /**
+     * @brief Setup and compile DirectML operator.
+     */
+    void LoadAssets(int m, int n, int k, DML_TENSOR_DATA_TYPE dataType);
+
+    /**
+     * @brief Initialize and tart the computation job.
+     * @return the elapsed time in ms.
+     */
+    template <typename T> double InitializeExecuteComputeOp(const int m, const int n, const int k);
+
+    /**
+     * @brief Wait until command completed.
+     */
+    void FlushCommandQueue();
+
+    /**
+     * @brief Close and execute command list, wait until command completed.
+     */
+    void CloseExecuteResetWait();
+
+    /**
+     * @brief Create a default buffer and upload data with the upload buffer.
+     * @param device the GPU device object.
+     * @param cmdList the GPU command list object.
+     * @param initData the data that need to upload.
+     * @param byteSize the size of data that need to upload.
+     * @param UploadBuffer the upload that use for upload data.
+     * @return a default buffer object.
+     */
+    Microsoft::WRL::ComPtr<ID3D12Resource> CreateDefaultBuffer(ID3D12Device *device, ID3D12GraphicsCommandList *cmdList,
+                                                               const void *initData, UINT64 byteSize,
+                                                               Microsoft::WRL::ComPtr<ID3D12Resource> &UploadBuffer);
+
+  private:
+    // Pipeline objects.
+    ComPtr<ID3D12Device> m_device;
+    ComPtr<ID3D12CommandAllocator> m_commandAllocator;
+    ComPtr<ID3D12CommandQueue> m_commandQueue;
+    ComPtr<ID3D12GraphicsCommandList> m_commandList;
+    ComPtr<IDMLDevice> m_dmlDevice;
+    ComPtr<IDMLCommandRecorder> m_dmlCommandRecorder;
+    ComPtr<IDMLCompiledOperator> m_dmlCompiledOperator;
+    ComPtr<IDMLBindingTable> m_bindingTable;
```

Comment:
why not set the default value nullptr like others?

Reply:
added the default nullptr for these